### PR TITLE
Pv2 82 move container bootstrapping code into common project

### PR DESCRIPTION
**Pull request by abroodie**
We are now using Autofac modules for registration.
ContainerFactory has now been created to make the bootstraping of the services a lot more consistent and to reduce the copy and paste nature of it.

Added ConfigHelper classes to help get config out of service fabric xml config files.

Have a look over the updated structure of the code and see what you think.

### CODE REVIEW

File: src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/PaymentHistoryRepositoryTest.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using System.Threading.Tasks;
 using AutoMapper;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.ServiceFabric.Data;
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/PaymentHistoryRepositoryTest.cs]: not required

File: src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/PaymentHistoryRepositoryTest.cs [hunk 2]
```diff
@@ -20,7 +21,7 @@ public class PaymentHistoryRepositoryTest
         private IPaymentHistoryRepository _repository;
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/PaymentHistoryRepositoryTest.cs]: naming conventions

File: src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/PaymentHistoryRepositoryTest.cs [hunk 3]
```diff
@@ -63,7 +64,9 @@ public async Task TestGetPaymentHistoryPopulatesCache()
             _paymentEntities.Add(new PaymentEntity {Id = Guid.NewGuid(), ApprenticeshipKey = apprenticeshipKey});
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/PaymentHistoryRepositoryTest.cs]: the apprenticeshipKey can be a constant

File: src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/PaymentHistoryRepositoryTest.cs [hunk 4]
```diff
@@ -87,7 +90,7 @@ public async Task TestGetPaymentHistoryUsesCache()
...
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/PaymentHistoryRepositoryTest.cs]: the apprenticeshipKey can be constant

Comment (davehowlett) [reply src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/PaymentHistoryRepositoryTest.cs]: there is another one in the final test

File: src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/ReliableCollectionCacheTest.cs [hunk 5]
```diff
@@ -7,9 +7,9 @@
 using Moq;
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/ReliableCollectionCacheTest.cs]: System, System.Text & SFA.DAS.Payments.RequiredPayments.Domain.Entities not used

File: src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/ReliableCollectionCacheTest.cs [hunk 6]
```diff
@@ -7,9 +7,9 @@
...
 using NUnit.Framework;
 using SFA.DAS.Payments.RequiredPayments.Application.Repositories;
-using SFA.DAS.Payments.RequiredPayments.ApprenticeshipPaymentsDueService;
 using SFA.DAS.Payments.RequiredPayments.Domain.Entities;
 using SFA.DAS.Payments.RequiredPayments.Model.Entities;
+using SFA.DAS.Payments.RequiredPayments.RequiredPaymentsService;
 
 namespace SFA.DAS.Payments.RequiredPayments.UnitTests.Application.Repositories
 {
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/ReliableCollectionCacheTest.cs]: naming conventions

Comment (davehowlett) [reply src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/ReliableCollectionCacheTest.cs]: a variable key can be made into a constant

File: src/SFA.DAS.Payments.RequiredPayments.UnitTests/Service/PayableEarningEventHandlerTest.cs [hunk 7]
```diff
@@ -18,8 +18,9 @@
 using SFA.DAS.Payments.ServiceFabric.Core;
 using Autofac;
 using ESFA.DC.Logging.Interfaces;
-using SFA.DAS.Payments.Core.LoggingHelper;
 using Autofac.Extras.Moq;
+using SFA.DAS.Payments.Application.Infrastructure.Logging;
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.RequiredPayments.UnitTests/Service/PayableEarningEventHandlerTest.cs]: SFA.DAS.Payments.ServiceFabric.Core not used

Comment (davehowlett) [reply src/SFA.DAS.Payments.RequiredPayments.UnitTests/Service/PayableEarningEventHandlerTest.cs]: nor is System.Collections.Generic

File: src/SFA.DAS.Payments.Service.Core/EndpointCommunicationBase.cs [hunk 8]
```diff
@@ -3,11 +3,10 @@
 using Autofac;
 using NServiceBus;
 using NServiceBus.Features;
-using SFA.DAS.Payments.Messages.Core;
 
 namespace SFA.DAS.Payments.ServiceFabric.Core
 {
-    public abstract class EndpointCommunicationBase<T> : IDisposable where T : IPaymentsMessage
+    public abstract class EndpointCommunicationBase : IDisposable
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.Service.Core/EndpointCommunicationBase.cs]: naming conventions

File: src/SFA.DAS.Payments.Service.Core/EndpointCommunicationListener.cs [hunk 9]
```diff
@@ -1,12 +1,12 @@
 ﻿using System.Threading;
 using System.Threading.Tasks;
 using Autofac;
+using SFA.DAS.Payments.Application.Infrastructure.Messaging;
 using SFA.DAS.Payments.Messages.Core;
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.Service.Core/EndpointCommunicationListener.cs]: not used and can be removed

File: src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/ReliableCollectionCacheTest.cs (hunk 6 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.RequiredPayments.UnitTests/Application/Repositories/ReliableCollectionCacheTest.cs]: The unit tests project is obsolete and in the process of being refactored into Domain.UnitTests and Application.UnitTests

File: src/SFA.DAS.Payments.RequiredPayments.UnitTests/Service/PayableEarningEventHandlerTest.cs (hunk 7 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.RequiredPayments.UnitTests/Service/PayableEarningEventHandlerTest.cs]: The unit tests project is obsolete and in the process of being refactored into Domain.UnitTests and Application.UnitTests

File: src/SFA.DAS.Payments.Service.Core/EndpointCommunicationBase.cs (hunk 8 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.Service.Core/EndpointCommunicationBase.cs]: Not sure what the issue is with this class name

**Pull request merged**